spdx/scanner.go

```diff
@@ -0,0 +1,71 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package spdx
+
+import (
+	"bufio"
+	"io"
+)
+
+type Scanner struct {
+	*bufio.Scanner
+}
+
+func spdxSplit(data []byte, atEOF bool) (advance int, token []byte, err error) {
+	//println(len(data), string(data), atEOF)
+	// skip WS
+	start := 0
+	for ; start < len(data); start++ {
+		if data[start] != ' ' && data[start] != '\n' {
+			break
+		}
+	}
+	if start == len(data) {
+		return start, nil, nil
+	}
+
+	// found (,)
+	switch data[start] {
+	case '(', ')':
+		return start + 1, data[start : start+1], nil
+	}
+
+	// found non-ws, non-(), must be a token
+	for i := start; i < len(data); i++ {
+		switch data[i] {
+		// token finished
```

Comment:
These two comments seem misplaced. Having them as first line inside the case would fix it. See below, though.

Reply:
Fixed this one too.